### PR TITLE
DAOS-12184 docker: Run DAOS application in a unique container

### PR DESCRIPTION
**Pull request by knard-intel**
# Description

Defines how to use docker containers for running an application accessing a DAOS file system.  The DAOS Agent service and the DAOS client are running in the same docker container.

Using DocOnly Pragma as the docker things are not tested by the CI.

# Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

# Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: utils/docker/examples/client/README.md [hunk 1]
```diff
@@ -215,3 +215,87 @@ $ dfuse --mountpoint="/home/<DAOS_CLIENT_UNAME>/mnt" --pool=tank --container=pos
...
+
+#### Running DAOS Docker Images with Docker Stack
+
+With Docker Stack the tarball of the certificates are managed as [Docker
+Secret](https://docs.docker.com/engine/swarm/secrets/).  Docker Secret is a swarm service allowing
+to securely store and access blob of data.  Recording a tarball containing the DAOS agent
+certificates could be done in the following way:
+```bash
+docker swarm init
+docker secret create daos_agent-certs <TARBALL PATH>
```
Comment (wiliamhuang) [inline utils/docker/examples/client/README.md:286]: Do you want to use <TARBALL PATH> or <DAOS_AGENT_CERTS_TXZ>? It's up to you.

**Review by wiliamhuang — APPROVED** [review]

File: utils/docker/examples/client/README.md (hunk 1 above)
Comment (knard-intel) [reply utils/docker/examples/client/README.md:286]: - [ ] Fix file name

Comment (knard-intel) [reply utils/docker/examples/client/README.md]: Fixed with commit 94325c116181e96f9685662051b275f0c3fc5bd5

**Comment by knard-intel** [comment]
@mark-olson , do you think you will be able to review this PR ?

**Review by mark-olson — APPROVED** [review]

**Pull request merged**